database.go

```diff
@@ -878,6 +908,24 @@ func (f *FieldCodec) DecodeFields(b []byte) map[uint8]interface{} {
 	return values
 }
 
+// DecodeFieldsWithNames decodes a byte slice into a set of field names and values
+func (f *FieldCodec) DecodeFieldsWithNames(b []byte) map[string]interface{} {
+	fields := f.DecodeFields(b)
+	m := make(map[string]interface{})
+	for id, v := range fields {
+		field := f.fieldsByID[id]
+		if field != nil {
+			m[field.Name] = v
+		}
+	}
+	return m
+}
+
+// FieldIDByName returns the field id by the name of the field. It will return a zero if not found
```

Comment:
I think the comment should read "....return a nil..." since `fieldsByName` is a map.